src/Symfony/Component/Encryption/EncryptionInterface.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Encryption;
+
+use Symfony\Component\Encryption\Exception\DecryptionException;
+use Symfony\Component\Encryption\Exception\EncryptionException;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ *
+ * @experimental in 5.3
+ */
+interface EncryptionInterface
+{
+    /**
+     * Generates new a Key to be used with encryption.
+     *
+     * Don't lose your private key and make sure to keep it a secret.
+     *
+     * @param string|null $secret to be used in symmetric encryption. A new secret
+     *                            is generated if none is provided.
+     *
+     * @throws EncryptionException
+     */
+    public function generateKey(string $secret = null): KeyInterface;
+
+    /**
+     * Get an encrypted version of the message.
+     *
+     * Symmetric encryption uses the same key to encrypt and decrypt a message.
+     * The key should be kept safe and should not be exposed to the public. Symmetric
+     * encryption should be used when you are not sending the encrypted message to
+     * anyone else.
+     *
+     * Example: You store a value on disk or in a cookie and don't want anyone else
+     * to read it.
+     *
+     * Symmetric encryption is in theory weaker than asymmetric encryption.
+     *
+     * <code>
+     *     $key = $encryption->generateKey();
+     *     $ciphertext = $encryption->encrypt('input', $key);
+     *     $message = $encryption->decrypt($ciphertext, $key);
+     * </code>
+     *
+     * @param string       $message plain text version of the message
+     * @param KeyInterface $myKey   a key that holds a string secret
+     *
+     * @return string the output
+     *
+     * @throws EncryptionException
+     */
+    public function encrypt(string $message, KeyInterface $myKey): string;
+
+    /**
+     * Get an encrypted version of the message that only the recipient can read.
+     *
+     * Asymmetric encryption uses a "key pair" ie a public key and a private key.
+     * It is safe to share your public key, but the private key should always be
+     * kept a secret.
+     *
+     * When Alice and Bob wants to communicate they share their public keys with
+     * each other. Alice will encrypt a message with bobs public key. When Bob
+     * receive the message, he will decrypt it with his private key.
+     *
```

Comment:
as you mentioned the word "recipient" above, maybe make it clear here who is who in this process, sender, recipient
because they both "receive" things, either a key or a message